pennylane/transforms/decompose.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A transform for decomposing quantum circuits into user defined gate sets. Offers an alternative to the more device-focused decompose transform.
+"""
+# pylint: disable=protected-access
+# pylint: disable=unnecessary-lambda-assignment
+
+import warnings
+from typing import Iterable
+
+import pennylane as qml
+from pennylane.transforms.core import transform
+
+
+def null_postprocessing(results):
+    """A postprocessing function returned by a transform that only converts the batch of results
+    into a result for a single ``QuantumTape``.
+    """
+    return results[0]
+
+
+@transform
+def decompose(tape, gate_set=None, max_expansion=None):
+    """Decomposes a quantum circuit into a user-specified gate set.
+
+    Args:
+        tape (QuantumScript or QNode or Callable): a quantum circuit.
+        gate_set (Iterable[Union[str, type]] or Callable[Operator, bool], optional): Decomposition gates defined by either (1) a gate set of operators or (2) a rule that they must follow.
+        Defaults to None. If ``None``, gate set defaults to all available operators given by ``~.pennylane.ops.__all__``.
```

Comment:
```suggestion
            Defaults to None. If ``None``, gate set defaults to all available operators given by ``~.pennylane.ops.__all__``.
```
<img width="744" alt="Screenshot 2024-10-08 at 4 50 38 PM" src="https://github.com/user-attachments/assets/3946361c-d315-4c29-843c-eff3252dd199">

Reply:
Oops!